### ISSUE TITLE
crf_decode_forward misses key RNN kwarg

### ISSUE THREAD

**Issue by tgsmith61591** [tensorflow/addons#2639]
In the `crf_decode_forward` method, the `RNN` constructor is missing the `zero_output_for_mask=True` kwarg.

https://github.com/tensorflow/addons/blob/f82a4a6f1d39e440cc028131b478056f3a55a5d7/tensorflow_addons/text/crf.py#L493

Without this argument, the provided `mask` will not zero out masked timesteps. Here is the [relevant documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RNN) (scroll down to the `args` section, as it's left out of the signature's documentation):

> **zero_output_for_mask**
  Boolean (default False). Whether the output should use zeros for the masked timesteps. Note that this field is only used when return_sequences is True and mask is provided. It can useful if you want to reuse the raw output sequence of the RNN without interference from the masked timesteps, eg, merging bidirectional RNNs.

**Comment by tgsmith61591** [comment]
Add a PR for this here: https://github.com/tensorflow/addons/pull/2642
